### PR TITLE
browser menu plural update

### LINKED ISSUE #14752: [BUG]: Plural words used in browser menu even when single card is selected
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

Select a single card in the Browser and open the menu.

### Expected behaviour

The menu options use singular words.

### Actual behaviour

Some menu options use plural words.

### Debug info

```text
AnkiDroid Version = 2.17alpha5 (4fc8af92973d8d502e0139d0dd5372415444ca1f)
```

### (Optional) Anything else you want to share?

_No response_

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by criticalAY**
## Purpose / Description
Correct string according to the number of cards

## Fixes
* Fixes #14752

## How Has This Been Tested?
Screenshot 2023-11-16 233948

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/menu/card_browser_multiselect.xml [hunk 1]
```diff
@@ -10,8 +10,7 @@
 
     <item
         android:id="@+id/action_delete_card"
-        android:icon="@drawable/ic_delete_white"
-        android:title="@string/card_browser_delete_card" />
+        android:icon="@drawable/ic_delete_white" />
```
Comment (david-allison) [inline AnkiDroid/src/main/res/menu/card_browser_multiselect.xml:13]: `tools:title` can be used to help designers (`Delete note(s)`)

File: AnkiDroid/src/main/res/values/07-cardbrowser.xml [hunk 2]
```diff
@@ -30,12 +30,22 @@
...
+    <plurals name="card_browser_delete_card">
+        <item quantity="one">Delete note</item>
+        <item quantity="other">Delete notes</item>
+    </plurals>
+    <string name="card_browser_suspend_card">Toggle suspend</string>
     <string name="card_browser_change_deck">Change deck</string>
-    <string name="card_browser_unsuspend_card">Unsuspend cards</string>
-    <string name="card_browser_mark_card">Mark notes</string>
-    <string name="card_browser_unmark_card">Unmark notes</string>
+<!--    <string name="card_browser_unsuspend_card">Unsuspend cards</string>-->
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/07-cardbrowser.xml:39]: No point in commenting it out - can be removed

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 3]
```diff
@@ -1897,16 +1904,22 @@ open class CardBrowser :
...
+                } else {
+                    resources.getQuantityString(R.plurals.card_browser_unsuspend_card, checkedCardCount())
+                }
                 setIcon(suspendIcon)
             }
-            val markTitle = if (hasUnmarked) R.string.card_browser_mark_card else R.string.card_browser_unmark_card
             val markIcon = if (hasUnmarked) R.drawable.ic_star_border_white else R.drawable.ic_star_white
             actionBarMenu.findItem(R.id.action_mark_card).apply {
-                title = getString(markTitle)
+                title = if (hasUnmarked) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1918]: (pending other reviewers): I'd much rather use Anki's "Toggle Suspend" and "Toggle Mark" here and remove this method

This code is complex and slow, and I don't feel it adds value to the end user

File: AnkiDroid/src/main/res/values/07-cardbrowser.xml [hunk 4]
```diff
@@ -30,12 +30,22 @@
     </plurals>
 
     <string name="card_browser_all_decks">All decks</string>
-    <string name="card_browser_delete_card">Delete notes</string>
-    <string name="card_browser_suspend_card">Suspend cards</string>
+    <plurals name="card_browser_delete_card">
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/07-cardbrowser.xml:33]: Let's fix the name here as well

File: AnkiDroid/src/main/res/values/07-cardbrowser.xml [hunk 5]
```diff
@@ -30,12 +30,22 @@
...
     <string name="card_browser_change_deck">Change deck</string>
-    <string name="card_browser_unsuspend_card">Unsuspend cards</string>
-    <string name="card_browser_mark_card">Mark notes</string>
-    <string name="card_browser_unmark_card">Unmark notes</string>
+<!--    <string name="card_browser_unsuspend_card">Unsuspend cards</string>-->
+    <plurals name="card_browser_unsuspend_card">
+        <item quantity="one">Unsuspend card</item>
+        <item quantity="other">Unsuspend cards</item>
+    </plurals>
+    <string name="card_browser_mark_card">Toggle Mark</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/07-cardbrowser.xml:44]: `CollectionManager.TR.browsingToggleMark()`

File: AnkiDroid/src/main/res/values/07-cardbrowser.xml [hunk 6]
```diff
@@ -30,12 +30,22 @@
     </plurals>
 
     <string name="card_browser_all_decks">All decks</string>
-    <string name="card_browser_delete_card">Delete notes</string>
-    <string name="card_browser_suspend_card">Suspend cards</string>
+    <plurals name="card_browser_delete_card">
+        <item quantity="one">Delete note</item>
+        <item quantity="other">Delete notes</item>
+    </plurals>
+    <string name="card_browser_suspend_card">Toggle suspend</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/07-cardbrowser.xml:37]: `CollectionManager.TR.browsingToggleSuspend()`

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 7]
```diff
@@ -922,6 +922,13 @@ open class CardBrowser :
                 resources.getQuantityString(R.plurals.card_browser_export_notes, checkedCardCount())
             }
         }
+        mActionBarMenu!!.findItem(R.id.action_delete_card).apply {
+            this.title = if (inCardsMode) {
+                resources.getQuantityString(R.plurals.card_browser_delete_card, checkedCardCount())
+            } else {
+                resources.getQuantityString(R.plurals.card_browser_delete_card, checkedCardCount())
+            }
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:930]: This looks to be the same on both branches

**Review by david-allison — CHANGES_REQUESTED** [review]
Thanks!

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt (hunk 3 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1918]: > "Toggle Suspend" and "Toggle Mark"

👍

**Comment by criticalAY** [comment]
Question : should we delete the string ? as it was also used to set title in XML file in the menu in case we use `CollectionManager.TR.browsingToggleMark()`

**Comment by BrayanDSO** [comment]
> should we delete the string ?

yes

> That would leave the title empty in the XML view

That's what `tools:title` serves for

> This looks to be the same on both branches
The code just above it is also doing so, should I simplify that too?

yes

**Comment by criticalAY** [comment]
> That's what tools:title serves for

No I meant if we use string from `TR` we wont have a string locally

**Comment by BrayanDSO** [comment]
> No I meant if we use string from TR we wont have a string locally

I still don't get what you mean. That we won't have XML previews for other languages if `tools:title` is being used?

**Comment by criticalAY** [comment]
> I still don't get what you mean. That we won't have XML previews for other languages if `tools:title` is being used?

I mean if we delete the string "Toggle mark" then we will have to hardcode the string as `tools:title` is it ok? as we wont have the "Toggle mark" string in the `values/strings`

**Comment by BrayanDSO** [comment]
I mean if we delete the string "Toggle mark" then we will have to hardcode the string as tools:title is it ok?

yes

**Comment by criticalAY** [comment]
@david-allison review requested! : )

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 8]
```diff
@@ -1897,16 +1904,22 @@ open class CardBrowser :
     private fun onSelectedCardsChecked(result: Pair<Boolean, Boolean>) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1904]: My proposal was to remove this method and the job creating it entirely and just keep 'Toggle Suspend' as a single title: https://github.com/ankidroid/Anki-Android/pull/14755#discussion_r1396418008

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt (hunk 8 above)
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt]: Removed!

**Comment by criticalAY** [comment]
I messed up the rebase but cleaned in the end, jobs need to be re-run though

**Review by david-allison — APPROVED** [review]
Perfect!

**Review by mikehardy — APPROVED** [review]
Nice - going to merge this and use it as a test of new string upload to crowdin with #11557 in place

**Pull request merged**